packages/app/src/components/SignInPage/SignInPage.tsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+
+import {
+  atlassianAuthApiRef,
+  bitbucketAuthApiRef,
+  bitbucketServerAuthApiRef,
+  configApiRef,
+  githubAuthApiRef,
+  gitlabAuthApiRef,
+  googleAuthApiRef,
+  microsoftAuthApiRef,
+  oktaAuthApiRef,
+  oneloginAuthApiRef,
+  useApi,
+  type SignInPageProps,
+} from '@backstage/core-plugin-api';
+
+import {
+  SignInPage as CCSignInPage,
+  ProxiedSignInPage,
+  type SignInProviderConfig,
+} from '@backstage/core-components';
+import { auth0AuthApiRef, oidcAuthApiRef, samlAuthApiRef } from '../../api';
+
+/**
+ * Key:
+ * string - Provider name.
+ *
+ * Value:
+ * SignInProviderConfig - Local sign-in provider configuration.
+ * string - Proxy sign-in provider configuration.
+ */
+const PROVIDERS = new Map<string, SignInProviderConfig | string>([
```

Comment:
Currently, it seems that only one provider can be configured and when multiple are configured, the provider higher up in this list of providers takes precedence over all the other providers. Since currently our "default" `app-config.yaml` files [include the github auth provider](https://github.com/janus-idp/backstage-showcase/blob/0d27cbbda96f6ae2bde21bb1540a3e326b84ac18/app-config.yaml#L200-L203) by default, the providers below github such as gitlab, google oidc, otka, saml and onelogin will never be configurable unless they completely remove the `app-config.yaml` and `app-config.example.yaml` that the image is loaded with.

Would we want to change up the behavior to allow all of the configured providers to render rather than only allowing one to appear?

Reply:
But I guess that wouldn't fix the issue of the github auth provider being rendered even if the user doesn't configure it.

Reply:
> Would we want to change up the behavior to allow all of the configured providers to render rather than only allowing one to appear?

@Zaperex No, we do not want to change this behavior because it would be a security concern. The more sign in methods we provide at the same time, the more attack vectors there are for bad actors. The platform engineer will need to provide their own `app-config` anyways so this should be a non-issue.